xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1281,14 +1281,13 @@ void DX::DeviceResources::SetHdrMetaData(DXGI_HDR_METADATA_HDR10& hdr10) const
       const double BP_1 = static_cast<double>(hdr10.BluePrimary[1]) / FACTOR_1;
       const double WP_0 = static_cast<double>(hdr10.WhitePoint[0]) / FACTOR_1;
       const double WP_1 = static_cast<double>(hdr10.WhitePoint[1]) / FACTOR_1;
-      const double Max_ML = static_cast<double>(hdr10.MaxMasteringLuminance) / FACTOR_2;
       const double min_ML = static_cast<double>(hdr10.MinMasteringLuminance) / FACTOR_2;
 
       CLog::LogF(LOGINFO,
                  "RP {:.3f} {:.3f} | GP {:.3f} {:.3f} | BP {:.3f} {:.3f} | WP {:.3f} "
-                 "{:.3f} | Max ML {:.0f} | min ML {:.4f} | Max CLL {} | Max FALL {}",
-                 RP_0, RP_1, GP_0, GP_1, BP_0, BP_1, WP_0, WP_1, Max_ML, min_ML,
-                 hdr10.MaxContentLightLevel, hdr10.MaxFrameAverageLightLevel);
+                 "{:.3f} | Max ML {} | min ML {:.4f} | Max CLL {} | Max FALL {}",
+                 RP_0, RP_1, GP_0, GP_1, BP_0, BP_1, WP_0, WP_1, hdr10.MaxMasteringLuminance,
```

Comment:
If this is wrong is also wrong in:
https://github.com/xbmc/xbmc/blob/912343ba35f1bff8ed66b882c97ba948fd3f1169/xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp#L543-L555

Is strange because years ago (when implemented) was checked with an HDMI analyzer and output metadata on HDMI was correct except for some decimal rounding errors that was fixed.

Then another fix in https://github.com/xbmc/xbmc/pull/19404 that prevents wrong metadata when FFmepg denominators not are the exact 50000 and 10000.

FFmpeg uses 10000 factors for both Max and Min ML this is sure.

I won't be able to look at this in detail until the weekend.

Reply:
I don't have that equipment and can't say for sure. MS page is clear about it though and it matches ST 2086 metadata definition.
Now that I try again I can't see a visual difference anymore, not sure why.
MS says in SetHDRMetaData() doc not to rely on it anymore...

Reply:
Tested HDMI HDR metadata with HDFury Arcana and found unexpected things...

Tested with video file:
```
Color primaries                          : BT.2020
Transfer characteristics                 : PQ
Matrix coefficients                      : BT.2020 non-constant
Mastering display color primaries        : Display P3
Mastering display luminance              : min: 0.0050 cd/m2, max: 4000 cd/m2
Maximum Content Light Level              : 349 cd/m2
Maximum Frame-Average Light Level        : 86 cd/m2
```

First tested on **Shield Pro 2019** for reference and because I remember perfectly that I had already tested the same when I bought Shield (December 2021):

![Shield](https://github.com/xbmc/xbmc/assets/58434170/5cdeaf29-c4a4-4f1e-9479-ff71ae61eb20)

Min ML should be 0.005 and is 0.0  (seems ignored). Note that in Android when HW acceleration is ON (MediaCodec Surface) this is handled by firmware and not with Kodi code probably this is a Shield firmware workaround or somewhat. At Dec 2021 all values was passed correctly.

**Windows 11:**

![Windows_11](https://github.com/xbmc/xbmc/assets/58434170/e6b8abeb-5198-4d6f-b9da-10a33ed1e1a1)

All values seems ignored and always (tested different sources) is Max ML 1499 and Max CLL 799. Probably Windows 11 currently ignores` SetHDRMetaData` as Microsoft mentions here:

> Warning
It is no longer recommended for apps to explicitly set HDR metadata on their swap chain using SetHDRMetaData. Windows does not guarantee that swap chain metadata is sent to the monitor, and monitors do not handle HDR metadata consistently.

https://learn.microsoft.com/en-us/windows/win32/api/dxgi1_5/nf-dxgi1_5-idxgiswapchain4-sethdrmetadata

Is very strange because picture looks good and very similar to Shield. 
Is Windows 11 tone mapping internally and generating fake meatadata of 1499 / 799? 

This also tested before (on Windows 10) and was working.....

It is all very puzzling at the moment and it is not possible to draw any clear conclusions 😒

Maybe next step is restore a old backup image of Windows 10 in Intel NUC and see what happens...

Reply:
That's the warning I was referring to.

Yes it's possible Windows is messing with things and "virtualizing" the tone mapping with the DWM.

But things changed over time. At one point, setting the colorspace was enough to turn on HDR on the TV, without the OS HDR switch set to on (installing old driver still does that I think).
My TV is reported by DXGI / Windows 10 with 1499 max lum. It's not the truth but aligned with your measure.

Be careful of additional processing by the TV (ex. "dynamic HDR" to fake dynamic metadata)

Stumbled across this: https://github.com/microsoft/DirectX-Graphics-Samples/issues/439
Don't know if that's still true or not, vendor dependent, ...
Maybe fullscreen + full screen exclusive helps.